### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by DirtyF** [jekyll/jekyll-admin#68]
A simple high-level roadmap would be be nice to share your vision on the project, at least to know what to expect and to be able to compare it with existing services like CloudCannon or similar open-source projects like https://github.com/gabriel-john/utterson that also aim at a functional UI for managing Jekyll-powered websites.

I guess the people at Github know exactly what they are doing and where they are going, maybe you will soon need help with prototyping UI or user testing before developing any new feature, we don't know. And if we don't know, we can not help.

**Comment by benbalter** [comment]
👋 @DirtyF. Thanks for opening this issue. The short answer, is that right now, it's just a GSoC project, scheduled to conclude next month. At that time, we're hoping to have the basic functionality in place (create/edit/delete pages, documents, config, and maybe data files), and hopefully push the first release.

From there, there's a handful of "nice to haves" like the ability to install plugins (or now themes), that make sense for immediate follow ups if @mertkahyaoglu is interested in continuing, or if other community members want to jump in. I can begin to open issues now for some of the things on our short-term wish list. That's about all we have in terms of a roadmap. 

GitHub has been mentoring the project, but we'd very much like it to be a community-owned project, and would love your help. There are efforts in https://github.com/jekyll/jekyll-admin/issues/67 to improve the documentation now that, as of this week, the server works front-to-back locally.

Beyond what's already been discussed, I personally think it'd be cool to package the front end as a Electron app to give a more native experience, but that's probably a while off as I suspect we'd want it to be a bit more polished before we start distributing it more broadly. And in the long run, there's nothing architecturally preventing a hosted instance (either the API or the front end, or both), but there's definitely some non-trivial technical challenges, at least in terms of scale, to overcome.

In short, we'd love to get more people involved. It's early in the project, things are just coming together this week where folks can begin to kick the tires, and I'll work on getting things documented so others can do just that, today and going forward.

**Comment by mertkahyaoglu** [comment]
> if @mertkahyaoglu is interested in continuing

I will definitely continue 🎉  and I have some other features in mind as well. In addition, I can take responsibility for creating Electron app and Atom Package.

**Comment by benbalter** [comment]
Opened issues for everything that's been discussed so far and created https://github.com/jekyll/jekyll-admin/milestone/1 with an Aug. 23 due date (end of GSoC).
